src/common/local_file_system.cpp

```diff
@@ -167,29 +167,45 @@ struct UnixFileHandle : public FileHandle {
 	};
 };
 
-static FileType GetFileTypeInternal(int fd) { // LCOV_EXCL_START
+static FileMetadata StatsInternal(int fd, const string &path) {
 	struct stat s;
 	if (fstat(fd, &s) == -1) {
-		return FileType::FILE_TYPE_INVALID;
+		throw IOException({{"errno", std::to_string(errno)}}, "Failed to get stats for file \"%s\": %s", path,
+		                  strerror(errno));
 	}
+
+	FileMetadata file_metadata;
+	file_metadata.file_size = s.st_size;
+	file_metadata.last_modification_time = Timestamp::FromEpochSeconds(s.st_mtime);
+
 	switch (s.st_mode & S_IFMT) {
 	case S_IFBLK:
-		return FileType::FILE_TYPE_BLOCKDEV;
+		file_metadata.file_type = FileType::FILE_TYPE_BLOCKDEV;
+		break;
```

Comment:
I made a mistake at first -- I missed the `break` so all file types are invalid, I will try to followup with compilation flags to detect.